### ISSUE TITLE
LLVM build output spew doubles the size CI logs.

### ISSUE THREAD

**Issue by eddyb** [rust-lang/rust#70468]
Out of the 12622 log lines in a recent `dist-x86_64-linux` build:
* 4843 of them are from the LLVM build itself
  * half of those are printing the file being compiled
  * the other half is this warning for every compiler invocation:
    ```
    clang-9: warning: argument unused during compilation: '-static-libstdc++' [-Wunused-command-line-argument]
    ```
* 1693 are from installing the LLVM build

So that's 6536 lines, over half of the log, that's noise from a successful LLVM build.

The result is slowed down log navigation, which can be frustrating when looking for an failure.

I'm not sure what the best way to quiet each of these things is, but I should mention that with `ninja` (and no `-static-libstdc++`), the only thing I see locally is the 1693 "Installing: ..." lines:
* the build itself is contained within one status line, assuming there are no warnings, but I'm not sure how `ninja` would behave on CI (on Windows CI it certainly still outputs one line per file)
* would be nice to get rid of those "Installing: ..." lines locally too

cc @Mark-Simulacrum

**Comment by jonhoo** [comment]
Just to leave a trail of breadcrumbs, there was an attempted fix for this in https://github.com/rust-lang/rust/pull/89983
